source/MRMesh/MRObjectSave.cpp

```diff
@@ -113,6 +116,47 @@ Polyline3 mergeToLines( const Object& object )
     return result;
 }
 
+// return shared models count
+int collectLinks( const Object& rootObject, const std::filesystem::path& scenePath, MapSharedObjectModelToLinkData& links, std::filesystem::path& sharedFolder )
+{
+    links.clear();
+
+    std::stack<const Object*> sceneGraphVisitedList;
+    sceneGraphVisitedList.push( &rootObject );
+
+    constexpr int maxFileNameLen = 12; // keep file names not too long to avoid hitting limit in some OSes
+    int countSharedFiles = 0;
+    while ( !sceneGraphVisitedList.empty() )
+    {
+        auto node = sceneGraphVisitedList.top();
+        sceneGraphVisitedList.pop();
+
+        if ( node->isAncillary() )
+            continue; // consider ancillary_ objects as temporary, not requiring saving
+
+        auto it = links.find( { node } );
+        if ( it == links.end() )
+            links[{ node }] = {};
+        else
+        {
+            if ( it->second.link.empty() )
+            {
+                // relative path to model in shared folder
+                it->second.link = fmt::format(
+                    "{}/{}_{}", cSharedFolderName, ++countSharedFiles,
```

Comment:
Please create folders using `/` operator of `std::filesystem::path`, which adds `\` on Windows